src/main/java/org/isf/menu/rest/UserController.java

```diff
@@ -358,5 +343,183 @@ public ResponseEntity<List<LitePermissionDTO>> retrievePermissionsByUsername(@Pa
+		if (userSettings == null || userSettings.isEmpty()) {
+		    LOGGER.info("UserSetting not found!");
+		    throw new OHAPIException(new OHExceptionMessage("UserSetting not found."));
+		}
+		UserSetting userSetting = userSettingManager.getUserSettingByUserNameConfigName(userName, configName);
+		if (userSetting == null) {
+		    LOGGER.info("No user settings for the user {}", userName);
+		    return ResponseEntity.status(HttpStatus.NO_CONTENT).body(null);
+		}
+		return ResponseEntity.ok(userSettingMapper.map2DTO(userSetting));
+	}
+
+	/**
+	 * Deletes a {@link UserSetting}.
+	 * 
+	 * @param id - the id of the userSetting {@link UserSetting} to delete.
+	 * @return {@code true} if the userSetting has been deleted, {@code false} otherwise.
+	 * @throws OHServiceException.
+	 */
+	@DeleteMapping(value = "/users/settings/{id}", produces = MediaType.APPLICATION_JSON_VALUE)
+	public ResponseEntity<Boolean> deleteUserSetting(@PathVariable(name = "id") int id) throws OHServiceException {
+		Optional<UserSetting> userSetting = userSettingManager.getUserSettingById(id);
+		final String ADMIN = "admin";
+		if (!userSetting.isPresent()) {
+		    LOGGER.info("No user settings with id {}", id);
+		    return ResponseEntity.status(HttpStatus.NO_CONTENT).body(null);
+		}
+		if (userSetting.get().getUser().equals(SecurityContextHolder.getContext().getAuthentication().getName()) ||
+						SecurityContextHolder.getContext().getAuthentication().getName().equals(ADMIN)) {
+			try {
+				 userSettingManager.deleteUserSetting(userSetting.get());
+		    	} catch (OHServiceException serviceException) {
+		            throw new OHAPIException(new OHExceptionMessage("UserSetting not deleted."));
+		    	}
+			return ResponseEntity.ok(true);
+		}
+		throw new OHAPIException(new OHExceptionMessage("Not allowed."));
+	}
+	
+	/**
+	 * Returns a {@link UserSettingDTO} of userSetting.
+	 * 
+	 * @return {@link UserSettingDTO} if the userSetting with configName dashboard exist, null otherwise.
+	 */
+	@GetMapping(value = "/user/settings/dashboard", produces = MediaType.APPLICATION_JSON_VALUE)
+	public ResponseEntity<UserSettingDTO> getUserSettingDashboard() throws OHServiceException {
+		LOGGER.info("Attempting to fetch the UserSetting of dashboard of the current user");
+		String userName = SecurityContextHolder.getContext().getAuthentication().getName();
+		String dashboard = "dashboard";
+		UserSetting userSetting = userSettingManager.getUserSettingByUserNameConfigName(userName, dashboard);
+		if (userSetting == null) {
+		    LOGGER.info("No dashboard settings for the current user {}", userName);
+		    return ResponseEntity.status(HttpStatus.NO_CONTENT).body(null);
+		} else {
+		    UserSettingDTO userSettingDashboard = userSettingMapper.map2DTO(userSetting);
+		    LOGGER.info("Found {} user settings", userSettingDashboard);
+		    return ResponseEntity.ok(userSettingDashboard);
+		}
```

Comment:
I think this is redundant, because we already have these:

- users/settings
- users/{userName}/settings/{configName}

Anyway should we allow non-ADMIN users to read other users settings and permissions?

Reply:
okay we can remove it.

Reply:
I have remove that endpoint